simulator/race_test.go

```diff
@@ -82,30 +131,19 @@ func TestRace(t *testing.T) {
 				t.Fatal(err)
```

Comment:
Here is the tricky part, about ```t.Fatal()``` in goroutine. I have read similar code before when I learn Golang and searched. There is a group discussion: ```Whenever t.Fatal is called in a goroutine, it is swallowed and lost. ``` (reference [https://groups.google.com/forum/#!topic/vault-tool/zNlHJLxoo3w](https://groups.google.com/forum/#!topic/vault-tool/zNlHJLxoo3w), also here [https://github.com/ipfs/go-ipfs/issues/2043](https://github.com/ipfs/go-ipfs/issues/2043)). One of their guys also used ```staticcheck``` to check their code. So, for double check, I also ran it against this file. Result is the same as theirs.  So, we can discuss about that. 
Line 119, 126, 131, 152

Reply:
Good catch, I've changed those to use `t.Error` instead.